Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2_test.py

```diff
@@ -1077,3 +1138,145 @@ def test_do_soap_request(self, mocker, requests_mock, integration_context, is_ne
         else:
             assert not new_token_mocker.called
         assert soap_mocker.call_count == len(http_call_attempt_results)
+
+    @pytest.mark.parametrize(
+        'xml_document, blacklisted_tags, expected_is_valid',
+        [
+            pytest.param(
+                '<DisplayField name="First Published">7194</DisplayField>',
+                [],
+                # Expected
+                True,
+                id='Valid',
+            ),
+            pytest.param(
+                '<ShowStatSummaries>false</ShowSummaries>',
+                [],
+                # Expected
+                False,
+                id='Mismatched tags',
+            ),
+            pytest.param(
+                '<ModuleCriteria><Module name="appname">5</Module></ModuleCriteria>',
+                ['ModuleCriteria'],
+                # Expected
+                False,
+                id='Blacklisted tag',
+            ),
+        ]
+    )
+    def test_is_valid_xml(self, xml_document: str, blacklisted_tags: list[str], expected_is_valid: bool):
+        """
+        Given:
+            - Case 1: A valid XML document as a string.
+            - Case 2: A malformed XML document (open and close tags mismatch).
+        When:
+            - Calling is_valid_xml.
+        Assert:
+            - Case 1: Ensure is_valid is True.
+            - Case 2: Ensure is_valid is False.
```

Comment:
```suggestion
            - Ensure the result matches the xml validity
```

Reply:
Fixed in commit cbeb616e474a7c5d67a4e15ac24b85286fcb58c8